### PR TITLE
Feat/display authorized user data

### PR DESCRIPTION
**Pull request by AniaPiWo**

### CODE REVIEW

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

### app/api/session/route.ts
- The code changes in this file look good.
- The addition of the check for `session` and returning a `NextResponse` with an error message and status code 401 is a good way to handle the case when the user is not logged in.
- No bugs or issues found.

### app/page.module.scss
- The code changes in this file look good.
- The addition of padding-top and padding-bottom to the `.container` class will provide consistent spacing in the application layout.
- No bugs or issues found.

### next.config.js
- The code changes in this file look good.
- The correction of the environment variable names from `GIHUB_ID` and `GIHUB_SECRET` to `GITHUB_ID` and `GITHUB_SECRET` is necessary for proper configuration.
- The addition of the `images` configuration for the `domains` property will allow images from the specified domain to be used in the application.
- No bugs or issues found.

### src/components/Filters/Filters.module.scss
- The code changes in this file look good.
- The removal of the padding-top and padding-bottom from the `.filters` class seems appropriate as it may not be necessary for the desired layout.
- No bugs or issues found.

### src/components/Header/Header.module.scss
- The code changes in this file look good.
- The adjustments made to the styles in the `.wrapper`, `.header`, `.buttons`, and `.githubAcc` classes seem appropriate for the desired layout and alignment.
- No bugs or issues found.

### src/components/Header/Header.tsx
- The code changes in this file look good.
- The addition of the logic to display the connected GitHub account information in the header when the user is authenticated and on specific pages is a good implementation of the task requirements.
- The use of `next/image` for displaying the GitHub avatar is a good practice for optimizing image loading and performance.
- No bugs or issues found.

Overall, the code changes align with the task requirements and there are no major bugs or issues found. The changes made in each file seem appropriate and follow best practices. Well done!

File: src/components/Header/Header.tsx [hunk 1]
```diff
@@ -5,22 +5,67 @@ import styles from './Header.module.scss'
...
   const { data: session, status } = useSession()
+  const name = session?.user?.name
+  const avatar = session?.user?.image
+  const pathname = usePathname()
+  const router = useRouter()
 
-  if (status === 'authenticated') {
+  if (
+    status === 'authenticated' &&
+    pathname === ('/my-profile' || '/create-profile')
```
Comment (Sumick) [inline src/components/Header/Header.tsx:23]: **nitpick:** This is not good solution, as it hardcodes pages urls that can change in time. My suggestion is to create different component, and `/my-profile`, and `/create-profile` can just have different layout component than homepage.

**Review by Sumick — CHANGES_REQUESTED** [review]
Let's split this Header into `AuthorizedHeader`, and `DefaultHeader` - then just create different layout component for Homepage, and these authorized routes, so we have it all clean and nice

File: app/api/session/route.ts [hunk 2]
```diff
@@ -4,7 +4,14 @@ import { NextResponse } from 'next/server'
 
 export async function GET(request: Request) {
   const session = await getServerSession(authOptions)
-  console.log(!!session)
+  console.log(session?.user?.name)
```
Comment (Sumick) [inline app/api/session/route.ts:7]: **nitpick:** Log to remove

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

### Task Goal
The goal of the task is to display authorized user data details in the application header.

### Requirements
- NextAuth must be used for authorization.
- Authorized user data should be displayed in the application header.

### Feedback
Based on the provided code changes, here is my feedback:

#### app/(profile)/create-profile/page.module.scss
- No changes were made in this file.

#### app/(profile)/create-profile/page.tsx
- The `LogOutBtn` component was imported.
- The `LogOutBtn` component was added to the return statement, allowing users to log out.

#### app/(profile)/layout.tsx
- This file was added.
- The `ProfileLayout` component was created to wrap the children components and provide a layout for the profile pages.

#### app/(profile)/my-profile/page.module.scss
- No changes were made in this file.

#### app/(profile)/my-profile/page.tsx
- The `LogOutBtn` component was imported.
- The `LogOutBtn` component was added to the return statement, allowing users to log out.

#### app/api/session/route.ts
- The code now logs the name of the user if the session exists.
- If the session does not exist, a 401 response is returned with a message indicating that the user is not logged in.

#### app/layout.tsx
- The `Header` component was removed from the import statement.
- The `Header` component was removed from the return statement.

#### app/page.module.scss
- The `container` class now has additional padding and align-items properties.

#### app/page.tsx
- The `Header` component was removed from the import statement.
- The `DefaultHeader` component was imported.
- The `DefaultHeader` component was added to the return statement, providing a header for the page.

#### next.config.js
- The environment variables `GIHUB_ID` and `GIHUB_SECRET` were renamed to `GITHUB_ID` and `GITHUB_SECRET`.
- The `images` property was added to configure the domains for image optimization.

#### src/components/DefaultHeader/DefaultHeader.module.scss
- This file was added.
- Styles for the default header component were defined.

#### src/components/DefaultHeader/DefaultHeader.tsx
- The component name was changed from `Header` to `DefaultHeader`.
- The import statements were updated accordingly.
- The `Header` component was replaced with the `DefaultHeader` component in the return statement.
- The "My profile" button now redirects to the `/my-profile` page.

#### src/components/Filters/Filters.module.scss
- The padding properties were removed from the `.container` class.

#### src/components/LogOutBtn/LogOutBtn.tsx
- This file was added.
- The `LogOutBtn` component was created to handle the log out functionality.

#### src/components/MyProfileHeader/MyProfileHeader.module.scss
- This file was renamed from `Header.module.scss`.
- The margin properties were removed from the `.logo` and `.frameLogin` classes.
- The `.buttons` class was removed.

#### src/components/MyProfileHeader/MyProfileHeader.tsx
- This file was added.
- The `MyProfileHeader` component was created to display the header for the profile pages.
- The user's name and avatar are displayed if available.

#### src/components/User/User.tsx
- The return statement was simplified to only display the session data as a JSON string.

### Overall Feedback
The code changes seem to align with the task requirements. The authorized user data is displayed in the application header, and NextAuth is used for authorization. The addition of the `LogOutBtn` component provides a way for users to log out. The new `ProfileLayout` component provides a layout for the profile pages.

The code changes also include some improvements, such as renaming environment variables, optimizing images, and refining the styling of various components.

Great job on implementing the required functionality and making these improvements!

Please let me know if you have any questions or need further assistance.

**Review by hansac91 — APPROVED** [review]

**Pull request merged**